server/src/main/java/org/elasticsearch/indices/recovery/RecoverySourceHandler.java

```diff
@@ -164,6 +183,10 @@ public RecoveryResponse recoverToTarget() throws IOException {
                 }
                 // we set this to 0 to create a translog roughly according to the retention policy
                 // on the target. Note that it will still filter out legacy operations with no sequence numbers
+
+                // DISCUSS: startingSeqNo = 0;
+                // Operations in translog is limited by 12h or 512MB, but there are no limit on non-deleted documents in Lucene history
+                // `startingSeqNo = 0` might replay a large number of operations.
                 startingSeqNo = 0;
```

Comment:
we might need to get the min-consecutive-seq-id from lucene here since we should have it? it's the min seqId in the policy, no?

Reply:
another option is to pass -1 and let the impl decide where to start.

Reply:
Here too I think we should just forgo the translog when we shift to lucene driven ops. That means that we need to set this to the local checkpoint in the lucene commint we copy. All previous ops will be part of the commit. The only reason why this is 0 is to create a "proper" history on the target in the form of a translog that is just as big as the local one. With a lucene backed history, this is not needed.

Reply:
I prefer to make this in a follow-up.